src/runtime/async_resource.rs

```diff
@@ -0,0 +1,102 @@
+use std::sync::{Arc, Weak};
```

Comment:
This module turned out not to be the right way to go, but it could be useful in other circumstances and getting it right was tricky, so I left it in.  Happy to remove it if you feel it's not worth the code space :)